### PR TITLE
Create .asf.yaml file

### PR DESCRIPTION
**Pull request by jcamachor**
## What is the purpose of the pull request

This PR creates the .asf.yaml file for the project. More information about the supported options can be found here: https://github.com/apache/infrastructure-asfyaml/blob/main/README.md
The current options basically preserve the behavior that was enabled before the transfer to the ASF organization.

### CODE REVIEW

File: .asf.yaml
Comment (ashvina) [inline .asf.yaml:25]: A wiki could be a helpful collaboration tool for XTable. For instance, it could be used to work together on the Community meeting agenda, or to share and discuss feature designs while they are in progress. This is hard to do with the dev mailing list or Slack.

File: .asf.yaml
Comment (ashvina) [inline .asf.yaml:32]: When squash and merge is disabled, should all the PRs squash the changes before merging?
This is the recommendation in the discussion link above for Apache Drill.

File: .asf.yaml
Comment (ashvina) [inline .asf.yaml:42]: Maybe dev list would be a better place for PRs and Issues notifications than commits? I'm not strongly opinionated on this. I just want to confirm if commits-list is deliberate.

File: .asf.yaml
Comment (zabetak) [reply .asf.yaml:32]: I think it can be done automatically via the rebase button. Otherwise it has to be done manually either by the PR author or by the maintainer who is going to merge the PR to main. Quite often I use the command line to squash and merge changes to the repo thus I am not using the GitHub UI much.

File: .asf.yaml
Comment (zabetak) [reply .asf.yaml:42]: The dev list should be as clean as possible and focused on human interaction instead of notifications. This allows newcomers to quickly blend in the community and outsiders to get a quick glance of what is happening in the project without getting overwhelmed by the sheer amount of notifications that comes when reviewing PRs/discussing issues etc. 

Additional mailing lists can be created if necessary but the current configuration where everything goes to commits@ is fine with me.

**Review by zabetak — APPROVED** [review]
LGTM, although difficult to test without merging!

File: .asf.yaml
Comment (jcamachor) [reply .asf.yaml:42]: AFAIK some projects maintain separate lists for commits and issues/PRs, while others argue that using mail filters with a single list is sufficient, as these emails are usually easy to categorize. I suggest we discuss and then decide on any changes through a vote, it is easy to create a new list if required.

**Pull request merged**
